tests/unit_tests/verRctNonSemanticsSimple.cpp

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) 2023, The Monero Project
+//
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without modification, are
+// permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of
+//    conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list
+//    of conditions and the following disclaimer in the documentation and/or other
+//    materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be
+//    used to endorse or promote products derived from this software without specific
+//    prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+// THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include <sstream>
+#include <mutex>
+
+#define IN_UNIT_TESTS // to access Blockchain::expand_transaction_2
+
+#include "gtest/gtest.h"
+#include "unit_tests_utils.h"
+
+#include "cryptonote_basic/cryptonote_format_utils.h"
+#include "cryptonote_core/blockchain.h"
+#include "file_io_utils.h"
+#include "misc_log_ex.h"
+#include "ringct/rctSigs.h"
+
+#define CACHED_SERIALIZATION_ASSUMPTION_SUBTEST(fieldmodifyclause)             \
+    do {                                                                       \
+        rct::rctSig modded_sig = get_original_sig();                           \
+        modded_sig.fieldmodifyclause;                                          \
+        const std::string mod_sig_hex = rct_sig_to_bin_hex_string(modded_sig); \
+        EXPECT_NE(original_sig_hex(), mod_sig_hex);                            \
+    } while (0);                                                               \
+
+namespace
```

Comment:
Instead of an anonymous `namespace` I'd encourage the `static` keyword to promote local reasoning (namespaces are less local than keywords). Not to mention all that non-functional whitespace from the indent... (indenting namespaces is a huge pet peeve for me).

Reply:
Removed the nonfunction whitespace and put `static` before the functions

Reply:
It looks better but what is the point of keeping the anonymous namespace in there?

Reply:
Avoiding linking conflicts and not polluting global namespace for tools like Doxygen

Reply:
1) linking conflicts mean there is a code smell (like introducing non-static functions into the global namespace, or badly named functions)
2) why would Doxygen be scraping `.cpp` files?

Reply:
> Avoiding linking conflicts and not polluting global namespace for tools like Doxygen

You won't get linking conflicts: anonymous namespace and `static` do exactly the same thing (for free functions).